modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_rule_builder.js

```diff
@@ -240,7 +242,32 @@ AUI.add(
 						instance._ruleClasses.set('fields', instance.getFields());
 						instance._ruleClasses.set('pages', instance.getPages());
 
-						instance._ruleClasses.render(rule);
+						if (!instance.ruleDraftEmpty(ruleDraft)) {
+							instance._ruleClasses.render(ruleDraft);
+						}
+						else {
+							instance._ruleClasses.render(rule);
+						}
+					},
+
+					ruleDraftEmpty: function(ruleDraft) {
+						var ruleDraftIsEmpty = true;
+
+						for (var i in ruleDraft) {
+							if (ruleDraft.hasOwnProperty(i)) {
+								ruleDraftIsEmpty = false;
+								break;
+							}
+						}
+						if (ruleDraft && ruleDraft.conditions) {
+							for (var j = 0; j < ruleDraft.conditions.length; j++) {
+								if (!ruleDraft.conditions[j].operands[0].value && !ruleDraft.conditions[j].operands.operator) {
```

Comment:
Hey @alinedoleron just for a better visualization, is a forms pattern group the if declarations with parentheses, so can you group these declarations? ex:` if((condition) && (instance._isBinaryCondition(index)) && (visible))`